src/bidiMapper/domains/context/browsingContextProcessor.ts

```diff
@@ -427,15 +431,64 @@ export class BrowsingContextProcessor {
     return {result: {}};
   }
 
-  process_input_performActions(
-    _params: Input.PerformActionsParameters
+  async process_input_performActions(
+    params: Input.PerformActionsParameters
   ): Promise<Message.EmptyResult> {
-    throw new Message.UnsupportedOperationException('Not implemented yet.');
+    const context = this.#browsingContextStorage.getContext(params.context);
+    const inputState = this.#inputStateManager.get(context.top);
+
+    const actionsByTick: ActionOption[][] = [];
+    for (const action of params.actions) {
+      switch (action.type) {
+        case Input.SourceActionsType.Pointer: {
+          action.parameters ??= {pointerType: Input.PointerType.Mouse};
+          action.parameters.pointerType ??= Input.PointerType.Mouse;
```

Comment:
Note: Interesting way of handling defaults

Reply:
See https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing